exotica_py/src/Exotica.cpp

```diff
@@ -0,0 +1,508 @@
+                        }
+                     }
+                }
+            }
+            return true;
+        }
+
+        bool load(handle src, bool)
+        {
+            return PyToInit(src.ptr(), value);
+        }
+
+        static PyObject* InitializerToTuple(const Initializer& src)
+        {
+            PyObject* dict = PyDict_New();
+            for(auto& prop : src.properties)
+            {
+                addPropertyToDict(dict, prop.first, prop.second);
+            }
+            return PyTuple_Pack(2,stringAsPy(src.getName()), dict);
+        }
+
+        static void addPropertyToDict(PyObject* dict, const std::string& name, const Property& prop)
+        {
+            if(prop.getType()=="std::string")
+            {
+                PyDict_SetItemString(dict, name.c_str(), py::cast(boost::any_cast<std::string>(prop.get())).ptr());
+            }
+            else if(prop.getType()=="int")
+            {
+                PyDict_SetItemString(dict, name.c_str(), py::cast(boost::any_cast<int>(prop.get())).ptr());
+            }
+            else if(prop.getType()=="long")
+            {
+                PyDict_SetItemString(dict, name.c_str(), py::cast(boost::any_cast<long>(prop.get())).ptr());
+            }
+            else if(prop.getType()=="double")
+            {
+                PyDict_SetItemString(dict, name.c_str(), py::cast(boost::any_cast<double>(prop.get())).ptr());
+            }
+            else if(prop.getType()=="Eigen::Matrix<double, -1, 1, 0, -1, 1>")
+            {
+                PyDict_SetItemString(dict, name.c_str(), py::cast(boost::any_cast<Eigen::VectorXd>(prop.get())).ptr());
+            }
+            else if(prop.getType()=="bool")
+            {
+                PyDict_SetItemString(dict, name.c_str(), py::cast(boost::any_cast<bool>(prop.get())).ptr());
+            }
+            else if(prop.getType()=="exotica::Initializer")
+            {
+                PyDict_SetItemString(dict, name.c_str(), InitializerToTuple(boost::any_cast<Initializer>(prop.get())));
+            }
+            else if(prop.isInitializerVectorType())
+            {
+                PyObject* vec = PyList_New(0);
+                for(Initializer& i : boost::any_cast<std::vector<Initializer>>(prop.get()))
+                {
+                    PyList_Append(vec, InitializerToTuple(i));
+                }
+                PyDict_SetItemString(dict, name.c_str(), vec);
+            }
+            else
+            {
+                HIGHLIGHT("Skipping unsupported type '"<<prop.getType()<<"'");
+            }
+        }        
+
+        static handle cast(Initializer src, return_value_policy /* policy */, handle /* parent */)
+        {
+            return handle(InitializerToTuple(src));
+        }
+    };
+}}
+
+
+PYBIND11_MODULE(exotica_py, module)
+{
+    //Setup::Instance();
+
+    module.doc() = "Exotica Python binding";
+
+    int argc = 0;
+    ros::init(argc, nullptr, "ExoticaPyDummyNode");
```

Comment:
To what?
```module.doc()``` is describing the module and node name is referring to just a dummy node required to run ROS commands internally. Open to suggestions